packages/flutter/lib/src/widgets/linked_text.dart

```diff
@@ -0,0 +1,759 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'text.dart';
+
+/// Signature for a callback that passes a [String] representing a link that has
+/// been tapped.
+typedef LinkTapCallback = void Function(String linkString);
+
+/// Signature for a function that builds an [InlineSpan] link.
+///
+/// The link displays [displayString] and links to [linkString] when tapped.
+/// These are distinct because sometimes a link may be split across multiple
+/// [TextSpan]s.
+///
+/// Creates a [GestureRecognizer] and returns it so that its lifecycle can be
+/// maintained by the caller.
+///
+/// {@template flutter.painting.LinkBuilder.recognizer}
+/// It's necessary to call [GestureRecognizer.dispose] on the returned
+/// recognizer when the owning widget is disposed. See [TextSpan.recognizer].
+/// When using [LinkedText], this is done automatically for any recognizers
+/// returned from a [LinkBuilder].
+/// {@endtemplate}
+typedef LinkBuilder = (InlineSpan, GestureRecognizer) Function(
```

Comment:
I am not entirely sure how this api would work. What if InlineSpan returned with multiple text span with recognizers?

Also I think you can traverse the inlinespan directly to find all the recognizer?

Reply:
also it is a weird ownership model. Not sure if possible, but I think the `RawGestureDetector.gestures` API would be better if you want the caller to pass the ownership to this widget

Reply:
Good call, I've gotten rid of the idea of returning recognizers like this and gone with a more typical approach like TextSpan.recognizer.  I tried to explain in more detail in https://github.com/flutter/flutter/pull/125927#issuecomment-1707027380.